File: Java/Day23.java
/*

Day 23: Implement a function that takes two integers as input and returns their greatest common divisor (GCD).

*/

// Program
import java.util.Scanner;

public class Day23{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the number a: ");
        int a = sc.nextInt();
        System.out.println("Enter the number b: ");
        int b = sc.nextInt();

        for(int i = Math.max(a,b); i > 0; i--){
            if(a%i == 0 && b%i == 0){
                System.out.println(String.format("gcd(%d,%d) = %d",a,b,i));
                break;
            }
        }

    }
}

/*

Input/Output:

Enter the number a: 
15
Enter the number b: 
30
gcd(15,30) = 15

*/

File: Python/Day23.py
"""

Day 23: Implement a function that takes two integers as input and returns their greatest common divisor (GCD)


"""

#Program

def hcf(a, b):
	if(b == 0):
		return a
	else:
		return hcf(b, a % b)

a = int(input("Enter the number:  "))
b = int(input("Enter the number: "))


print(f"The gcd of {a} and {b} is : ", end="")
print(hcf(a, b))




"""

Input and Output:

Enter the number:  60
Enter the number: 48
The gcd of 60 and 48 is : 12


"""